mcs/class/System.Drawing/System.Drawing/gdipFunctions.cs

```diff
@@ -979,22 +988,22 @@ static internal void CheckStatus (Status status)
 		internal static extern Status GdipBitmapSetPixel (IntPtr bmp, int x, int y, int argb);
 
 		// Image functions
-		[DllImport(GdiPlus, CharSet=CharSet.Auto)]
+		[DllImport(GdiPlus, CharSet=GdiPlusCharSet)]
 		internal static extern Status GdipLoadImageFromFile ( [MarshalAs(UnmanagedType.LPWStr)] string filename, out IntPtr image );
```

Comment:
You're right:
- All strings have an explicit attribute on them forcing `LPWStr` marshaling, so it seems the `CharSet` attribute is redundant.
- Most (but not all) functions on the C side only expose a Unicode equivalent - e.g. [`GdipPrivateAddFontFile`](https://github.com/mono/libgdiplus/blob/7d30e92a6cfd6882cd57efeb7fb7b21b31ae2736/src/fontcollection.h#L41) only takes a `GDIPCONST WCHAR*` and there's no `GdipPrivateAddFontFileA`/`GdipPrivateAddFontFileW` variation, so ANSI would never be correct.
- I'm guessing Mono no longer supports Windows 95, 98 or Me so backwards Windows compatibility shouldn't be an issue, either

So there are two options:
- Remove the `CharSet` property from the `DllImport` attribute alltogether
- Using `Unicode` everywhere (since that's what libgdiplus expects)

Let me know which one you prefer and I'll update the PR.